sw/device/lib/dif/dif_aes_unittest.cc

```diff
@@ -182,13 +193,257 @@ class DecryptTest : public AesTestInitialized {
+  SetExpectedConfig(AES_CTRL_SHADOWED_KEY_LEN_VALUE_AES_128,
                     AES_CTRL_SHADOWED_MODE_VALUE_AES_ECB,
-                    AES_CTRL_SHADOWED_OPERATION_VALUE_AES_DEC);
-  setExpectedKey(kKey_, 8);
+                    AES_CTRL_SHADOWED_OPERATION_VALUE_AES_ENC,
+                    /*manual_op=*/true,
+                    /*force_zero_masks=*/false);
+  SetExpectedKey(kKey_, 8);
 
   EXPECT_EQ(dif_aes_start(&aes_, &transaction, kKey_, NULL), kDifOk);
 }
 
+// Zero masking
+class ZeroMaskingTest : public AesTestInitialized {
+ protected:
+  ZeroMaskingTest() { transaction.masking = kDifAesMaskingForceZero; }
+};
+
+TEST_F(ZeroMaskingTest, start) {
+  EXPECT_READ32(AES_STATUS_REG_OFFSET, 1);
+  SetExpectedConfig(AES_CTRL_SHADOWED_KEY_LEN_VALUE_AES_128,
+                    AES_CTRL_SHADOWED_MODE_VALUE_AES_ECB,
+                    AES_CTRL_SHADOWED_OPERATION_VALUE_AES_ENC,
+                    /*manual_op=*/false,
+                    /*force_zero_masks=*/true);
+  SetExpectedKey(kKey_, 8);
+
+  EXPECT_EQ(dif_aes_start(&aes_, &transaction, kKey_, NULL), kDifOk);
+}
+
+// Alert test.
+class AlertTest : public AesTestInitialized {};
+
+TEST_F(AlertTest, RecovCtrlUpdateErr) {
+  EXPECT_WRITE32(AES_ALERT_TEST_REG_OFFSET,
+                 {{AES_ALERT_TEST_RECOV_CTRL_UPDATE_ERR_BIT, true},
+                  {AES_ALERT_TEST_FATAL_FAULT_BIT, false}});
+
+  EXPECT_EQ(dif_aes_alert_force(&aes_, kDifAesAlertRecovCtrlUpdateErr), kDifOk);
+}
+
+TEST_F(AlertTest, AlertFatalFault) {
+  EXPECT_WRITE32(AES_ALERT_TEST_REG_OFFSET,
+                 {{AES_ALERT_TEST_RECOV_CTRL_UPDATE_ERR_BIT, false},
+                  {AES_ALERT_TEST_FATAL_FAULT_BIT, true}});
+
+  EXPECT_EQ(dif_aes_alert_force(&aes_, kDifAesAlertFatalFault), kDifOk);
+}
+
+// Data in
+class Data : public AesTestInitialized {
+ protected:
+  const dif_aes_data_t data_ = {
+      .data = {0xA55AA55A, 0xA55AA55A, 0xA55AA55A, 0xA55AA55A}};
+};
+
+TEST_F(Data, DataIn) {
+  EXPECT_READ32(AES_STATUS_REG_OFFSET, {
+                                           {AES_STATUS_INPUT_READY_BIT, true},
+                                       });
+
+  for (uint32_t i = 0; i < sizeof(data_.data) / sizeof(data_.data[0]); i++) {
+    ptrdiff_t offset = AES_DATA_IN_0_REG_OFFSET + (i * sizeof(uint32_t));
+    EXPECT_WRITE32(offset, data_.data[i]);
+  }
+
+  EXPECT_EQ(dif_aes_load_data(&aes_, data_), kDifOk);
+}
+
+TEST_F(Data, DataOut) {
+  EXPECT_READ32(AES_STATUS_REG_OFFSET, {
+                                           {AES_STATUS_INPUT_READY_BIT, true},
+                                           {AES_STATUS_OUTPUT_VALID_BIT, true},
+                                       });
+
+  for (uint32_t i = 0; i < ARRAYSIZE(data_.data); ++i) {
+    ptrdiff_t offset = AES_DATA_OUT_0_REG_OFFSET + (i * sizeof(uint32_t));
+    EXPECT_READ32(offset, data_.data[i]);
+  }
+
+  dif_aes_data_t out;
+  EXPECT_EQ(dif_aes_read_output(&aes_, &out), kDifOk);
+
+  EXPECT_THAT(out.data, testing::ElementsAreArray(data_.data));
```

Comment:
It's common style to include a `using` statement for all matchers that come from the `testing::` namespace.